packages/arch3-core/src/signingarchwayclient.spec.ts

```diff
@@ -388,28 +388,93 @@ describe('SigningArchwayClient', () => {
+        const [wallet, accounts] = await getWalletWithAccounts();
+        const client = await SigningArchwayClient.connectWithSigner(archwayd.endpoint, wallet, clientOptions);
 
-      const rewardsAddress = accounts[1].address;
+        const contractAddress = contracts.voter.addresses[2];
+        const ownerAddress = accounts[2].address;
+        const rewardsAddress = accounts[5].address;
 
-      const result = await client.withdrawContractRewards(rewardsAddress, 0, 'auto');
+        const metadata: ContractMetadata = {
+          contractAddress,
+          ownerAddress,
+          rewardsAddress,
+          withdrawToWallet: true
+        };
+        const result = await client.setContractMetadata(ownerAddress, metadata, 'auto');
 
-      expect(result).toMatchObject({
-        height: expect.any(Number),
-        transactionHash: expect.any(String),
-        gasWanted: expect.any(Number),
-        gasUsed: expect.any(Number),
-        rewardsAddress: rewardsAddress,
-        rewards: expect.arrayContaining([]),
+        expect(result).toMatchObject({
+          height: expect.any(Number),
+          transactionHash: expect.any(String),
+          gasWanted: expect.any(Number),
+          gasUsed: expect.any(Number),
+          metadata,
+        });
+        expect(result.logs).not.toHaveLength(0);
+        expect(result.events).not.toHaveLength(0);
+
+        client.disconnect();
       });
-      expect(result.logs).not.toHaveLength(0);
-      expect(result.events).not.toHaveLength(0);
 
-      client.disconnect();
+      it('doesn\'t generate RewardRecord entries when withdrawToWallet is enabled', async () => {
+        const [wallet, accounts] = await getWalletWithAccounts();
+        const client = await SigningArchwayClient.connectWithSigner(archwayd.endpoint, wallet, clientOptions);
+
+        const contractAddress = contracts.voter.addresses[2];
+        const rewardsAddress = accounts[5].address;
+
+        /* eslint-disable camelcase, @typescript-eslint/naming-convention */
+        const msg = {
+          new_voting: {
+            name: 'test_voting_rewards',
+            vote_options: ['yes', 'no'],
+            duration: 10000000000,
+          }
+        };
+        /* eslint-enable camelcase, @typescript-eslint/naming-convention */
+        await client.execute(rewardsAddress, contractAddress, msg, 'auto', undefined, coins(10, archwayd.denom));
+
+        const result = await client.withdrawContractRewards(rewardsAddress, 0, 'auto');
+
+        expect(result).toMatchObject({
+          height: expect.any(Number),
+          transactionHash: expect.any(String),
+          gasWanted: expect.any(Number),
+          gasUsed: expect.any(Number),
+          rewardsAddress: rewardsAddress,
+          rewards: expect.arrayContaining([]),
+        });
+        expect(result.logs).not.toHaveLength(0);
+        expect(result.events).not.toHaveLength(0);
+
+        client.disconnect();
+      });
```

Comment:
This test isn't necessary because it's checking a protocol functionality rather than something specific to `arch3.js`.